### PR TITLE
Add Repositories to Workspace

### PR DESCRIPTION
**Pull request by Vayras**
this PR is will add the feature to add repositories to workspace, finalising some design changes
#519

### CODE REVIEW

**Comment by elraphty** [comment]
@Vayras can you finish this today?

**Comment by Vayras** [comment]
> @Vayras can you finish this today?

@elraphty just waiting on the delete and modify endpoints

File: src/store/main.ts [hunk 1]
```diff
@@ -35,6 +35,12 @@ export interface Tribe {
...
 }
 
+export interface Repository {
+  workspace_uuid: string;
+  name: string;
+  url: string;
+}
+
 export interface Bot {
   id?: number;
```
Comment (elraphty) [inline src/store/main.ts:45]: Move this to interface.ts

**Comment by elraphty** [comment]
@Vayras please add a cypress test

**Comment by Vayras** [comment]
@elraphty the test will require the modify and delete endpoint are they ready?

File: src/pages/index.tsx [hunk 2]
```diff
@@ -48,6 +49,10 @@ const modeDispatchPages: Record<AppMode, () => React.ReactElement> = {
           <Route path="/workspace/:uuid">
             <WorkspaceMission />
           </Route>
+          <Route path="/workspaces/:uuid">
+            <AddRepos />
```
Comment (elraphty) [inline src/pages/index.tsx:53]: You do not need to create a new route for repositories, the repositories are to be in this file according to the design 
```sh
 <Route path="/workspace/:uuid">
       <WorkspaceMission />
 </Route>
```

File: src/store/main.ts (hunk 1 above)
Comment (Vayras) [reply src/store/main.ts:45]: sure will do

File: src/pages/index.tsx (hunk 2 above)
Comment (Vayras) [reply src/pages/index.tsx:53]: okay will move it to that

**Comment by Vayras** [comment]
@elraphty added the cypress test

**Comment by elraphty** [comment]
@Vayras this test is failing `43_rolesFundWorkspace.cy.ts `, please take a look

**Comment by Vayras** [comment]
> @Vayras this test is failing `43_rolesFundWorkspace.cy.ts `, please take a look

@elraphty working on it

**Pull request merged**